src/popupmenu.c

```diff
@@ -416,6 +416,109 @@ pum_under_menu(int row, int col, int only_redrawing)
 	    && col < pum_col + pum_width + pum_scrollbar;
 }
 
+/*
+ * displays text on the popup menu with specific attributes.
+ */
+    static void
+pum_screen_put_with_attr(int row, int col, char_u *text, int textlen, int attr)
+{
+    int		i;
+    int		leader_len;
+    int		char_len;
+    int		cells;
+    int		new_attr;
+    char_u	*rt_leader = NULL;
+    char_u	*match_leader = NULL;
+    char_u	*ptr = text;
+    garray_T	*ga = NULL;
+    char_u	*leader = ins_compl_leader();
+    int		in_fuzzy = (get_cot_flags() & COT_FUZZY) != 0;
```

Comment:
How about this, to not having to figure out extra highlighting when not required?

```suggestion
    if (highlight_attr[HLF_PMSI] == highlight_attr[HLF_PSI] &&
       highlight_attr[HLF_PMNI] == highlight_attr[HLF_PNI])
    {
       screen_puts_len(text, textlen, row, col, attr);
       return;
    }
```

Reply:
this will cause Match highlight only work with fuzzy.

Reply:
I have do some clean and update the description of Match relate highlight. they are only make sense when in fuzzy.